nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+import static org.sonatype.repository.helm.internal.database.HelmProperties.ICON;
+import static org.sonatype.repository.helm.internal.database.HelmProperties.MAINTAINERS;
+import static org.sonatype.repository.helm.internal.database.HelmProperties.NAME;
+import static org.sonatype.repository.helm.internal.database.HelmProperties.SOURCES;
+import static org.sonatype.repository.helm.internal.database.HelmProperties.VERSION;
+
+/**
+ * Build index.yaml file for Helm Hosted
+ *
+ * @since 1.0.15
+ */
+@Named
+@Singleton
+public class CreateIndexServiceImpl
+    extends ComponentSupport
+    implements CreateIndexService
+{
+  private static final String API_VERSION = "v1";
+
+  private static final String INDEX_YAML_CONTENT_TYPE = "text/x-yaml";
+
+  private final YamlParser yamlParser;
+
+  @Inject
+  public CreateIndexServiceImpl(final YamlParser yamlParser) {
+    this.yamlParser = checkNotNull(yamlParser);
+  }
+
+  @Nullable
+  public Content buildIndexYaml(final Repository repository) {
+    HelmContentFacet helmFacet = repository.facet(HelmContentFacet.class);
+
+    ChartIndex index = new ChartIndex();
+
+    for (Asset asset : helmFacet.browseComponentAssets(AssetKind.HELM_PACKAGE)) {
+      parseAssetIntoChartEntry(index, asset);
+    }
+
+    index.setApiVersion(API_VERSION);
+    index.setGenerated(new DateTime());
+
+    return new Content(new StringPayload(yamlParser.getYamlContent(index), INDEX_YAML_CONTENT_TYPE));
+  }
+
+  private void parseAssetIntoChartEntry(final ChartIndex index, final Asset asset) {
+    NestedAttributesMap formatAttributes = asset.attributes().child(HelmFormat.NAME);
+    if (formatAttributes != null) {
+      ChartEntry chartEntry = new ChartEntry();
+      chartEntry.setName(formatAttributes.get(NAME.getPropertyName(), String.class));
+      chartEntry.setVersion(formatAttributes.get(VERSION.getPropertyName(), String.class));
+      chartEntry.setDescription(formatAttributes.get(DESCRIPTION.getPropertyName(), String.class));
+      chartEntry.setIcon(formatAttributes.get(ICON.getPropertyName(), String.class));
+      chartEntry.setCreated(new DateTime(asset.created().toString()));
+
+      @SuppressWarnings("unchecked")
+      List<Map<String, String>> maintainers = formatAttributes.get(MAINTAINERS.getPropertyName(), List.class);
+      chartEntry.setMaintainers(maintainers);
+      chartEntry.setAppVersion(formatAttributes.get(APP_VERSION.getPropertyName(), String.class));
+      AssetBlob blob = asset.blob().orElse(null);
+      chartEntry.setDigest(blob != null ? blob.checksums().get("sha256") : null);
+      createListOfRelativeUrls(formatAttributes, chartEntry);
+      chartEntry.setSources(formatAttributes.get(SOURCES.getPropertyName(), List.class));
+      index.addEntry(chartEntry);
+    }
+  }
+
+  private void createListOfRelativeUrls(final NestedAttributesMap formatAttributes, final ChartEntry chartEntry) {
+    List<String> urls = new ArrayList<>();
+    urls.add(String.format("%s-%s.tgz",
```

Comment:
Minor: we used to create constants for string formatting in other formats

Reply:
just copied method from `CreateIndexServiceImpl` class for Orient